src/DFG.cpp

```diff
@@ -27,16 +27,49 @@ DFG::DFG(Function& t_F, list<Loop*>* t_loops, bool t_targetFunction,
 //  tuneForBitcast();
 //  tuneForLoad();
   if (t_heterogeneity) {
+    //calculateCycles();
+    // *************** testDualIssue *****************
+    // list<string>* targetPattern1 = new list<string>();
+    // targetPattern1->push_back("br");
+    // targetPattern1->push_back("phi");
+    // targetPattern1->push_back("add");
+    // targetPattern1->push_back("icmp");
+    // combineForIter(targetPattern1);
+    // list<string>* targetPattern1 = new list<string>();
+    // targetPattern1->push_back("phi");
+    // targetPattern1->push_back("icmp");
+    // targetPattern1->push_back("or");
+    // targetPattern1->push_back("br");
+    // targetPattern1->push_back("phi");
+    // combineForIter(targetPattern1);
+    // list<string>* targetPattern1 = new list<string>();
+    // targetPattern1->push_back("br");
+    // targetPattern1->push_back("xor");
+    // targetPattern1->push_back("phi");
+    // targetPattern1->push_back("icmp");
+    // targetPattern1->push_back("br");
+    // combineForIter(targetPattern1);
+    // list<string>* targetPattern1 = new list<string>();
+    // targetPattern1->push_back("add");
+    // targetPattern1->push_back("phi");
+    // targetPattern1->push_back("or");
+    // targetPattern1->push_back("br");
+    // combineForUnroll(targetPattern1);
+    // combine("icmp", "br");
+    // combine("getelementptr", "load");
+    // tuneForPattern();
+    // ESCORT();
     calculateCycles();
```

Comment:
Looks like our original code was wrong? the `calculateCycles()` happen before `combine()`, wdyt?